core/src/test/java/org/elasticsearch/index/query/FuzzyQueryBuilderTest.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.query;
+
+import org.apache.lucene.index.Term;
+import org.apache.lucene.search.FuzzyQuery;
+import org.apache.lucene.search.MultiTermQuery;
+import org.apache.lucene.search.Query;
+import org.elasticsearch.common.ParseFieldMatcher;
+import org.elasticsearch.common.lucene.BytesRefs;
+import org.elasticsearch.common.unit.Fuzziness;
+import org.elasticsearch.index.mapper.MappedFieldType;
+import org.elasticsearch.index.query.support.QueryParsers;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.junit.Test;
+
+import java.io.IOException;
+
+import static org.hamcrest.Matchers.is;
+
+public class FuzzyQueryBuilderTest extends BaseQueryTestCase<FuzzyQueryBuilder> {
+
+    private static final String[] REWRITE_METHOD = new String[]{"constant_score", "scoring_boolean", "constant_score_boolean", 
+            "top_terms_1", "top_terms_boost_1", "top_terms_blended_freqs_1"};
+
+    private static final String[] MAPPED_FIELD_NAME = new String[] { INT_FIELD_NAME, DOUBLE_FIELD_NAME, DATE_FIELD_NAME, 
+            STRING_FIELD_NAME };
+    
+    @Override
+    protected FuzzyQueryBuilder doCreateTestQueryBuilder() {
+        FuzzyQueryBuilder query;
+
+        // mapped or unmapped field
+        if (randomBoolean()) {
+            Object text = randomAsciiOfLengthBetween(1, 50);
+            String fieldName = randomFrom(MAPPED_FIELD_NAME);
+            if (fieldName.equals(INT_FIELD_NAME)) {
+                text = randomIntBetween(0, 10);
+            } else if (fieldName.equals(DOUBLE_FIELD_NAME)) {
+                text = randomDouble() * 10;
+            } else if (fieldName.equals(DATE_FIELD_NAME)) {
+                text = new DateTime(System.currentTimeMillis(), DateTimeZone.UTC);
+            }
+            query = new FuzzyQueryBuilder(fieldName, text);
+        } else {
+            query = new FuzzyQueryBuilder(randomAsciiOfLengthBetween(1, 10), randomAsciiOfLengthBetween(1, 50));
```

Comment:
I think it would be important to test unmapped fields against other types besides String. Thats what BaseTermQueryTestCase does in doCreateTestQueryBuilder. I think we should do exactly the same here no?